platform/src/main/java/org/stellar/anchor/platform/event/EventProcessorManager.java

```diff
@@ -0,0 +1,163 @@
+  public static final String CLIENT_STATUS_CALLBACK_EVENT_PROCESSOR_NAME_PREFIX =
+      "client-status-callback-";
+  public static final String CALLBACK_API_EVENT_PROCESSOR_NAME = "callback-api";
+  private final EventProcessorConfig eventProcessorConfig;
+  private final CallbackApiConfig callbackApiConfig;
+  private final ClientsConfig clientsConfig;
+  private final EventService eventService;
+
+  private final List<EventProcessor> processors = new ArrayList<>();
+
+  public EventProcessorManager(
+      EventProcessorConfig eventProcessorConfig,
+      CallbackApiConfig callbackApiConfig,
+      ClientsConfig clientsConfig,
+      EventService eventService) {
+    this.eventProcessorConfig = eventProcessorConfig;
+    this.callbackApiConfig = callbackApiConfig;
+    this.clientsConfig = clientsConfig;
+    this.eventService = eventService;
+  }
+
+  @PostConstruct
+  public void start() {
+    if (eventProcessorConfig.getCallbackApiRequest().isEnabled()) {
+      // Create a processor for the callback API handler
+      processors.add(
+          new EventProcessor(
+              CALLBACK_API_EVENT_PROCESSOR_NAME,
+              EventQueue.TRANSACTION,
+              new CallbackApiHandler(callbackApiConfig)));
+      // Create a processor of the client status callback handler for each client defined in the
+      // clientsConfig
+      if (eventProcessorConfig.getClientStatusCallback().isEnabled()) {
+        clientsConfig
+            .getClients()
+            .forEach(
+                (clientConfig) -> {
+                  String processorName = null;
+                  switch (clientConfig.getType()) {
+                    case CUSTODIAL:
+                      processorName =
+                          CLIENT_STATUS_CALLBACK_EVENT_PROCESSOR_NAME_PREFIX
+                              + clientConfig.getSigningKey();
+                      break;
+                    case NONCUSTODIAL:
+                      processorName =
+                          CLIENT_STATUS_CALLBACK_EVENT_PROCESSOR_NAME_PREFIX
+                              + clientConfig.getDomain();
+                      break;
+                    default:
+                      errorF("Unknown client type: {}", clientConfig.getType());
+                  }
+                  processors.add(
+                      new EventProcessor(
+                          processorName,
+                          EventQueue.TRANSACTION,
+                          new ClientStatusCallbackHandler(clientConfig)));
+                });
+      }
+    }
+
+    // Start all the processors
+    processors.forEach(EventProcessor::start);
+  }
+
+  @PreDestroy
+  public void stop() {
+    for (EventProcessor processor : processors) {
+      try {
+        processor.stop();
+      } catch (AnchorException ex) {
+        errorF("Failed to shutdown event processor: {}}", processor.getClass());
+        errorEx(ex);
+      }
+    }
+  }
+
+  class EventProcessor implements Runnable {
```

Comment:
Would we need to initialize two queues for the SQS implementation, one for status callback and the other business server events?

Reply:
In SQS, I think we will need to have a pre-processor that dispatches the incoming queue to multiple queues. Then one processor for each queue.

Reply:
Will this pre-processor be implemented in `EventService.Session`?

Reply:
I am thinking starting the pre-processor in the bean construction.